src/vorta/views/repo_add_dialog.py

```diff
@@ -27,28 +27,27 @@ def __init__(self, parent=None):
         self.saveButton = self.buttonBox.button(QDialogButtonBox.StandardButton.Ok)
         self.saveButton.setText(self.tr("Add"))
 
+        self.passwordInput = PasswordInput()
+        self.passwordInput.add_form_to_layout(self.repoDataFormLayout)
+
         self.buttonBox.rejected.connect(self.close)
         self.buttonBox.accepted.connect(self.run)
         self.chooseLocalFolderButton.clicked.connect(self.choose_local_backup_folder)
         self.useRemoteRepoButton.clicked.connect(self.use_remote_repo_action)
         self.repoURL.textChanged.connect(self.set_password)
-        self.passwordLineEdit.textChanged.connect(self.password_listener)
-        self.confirmLineEdit.textChanged.connect(self.password_listener)
-        self.encryptionComboBox.activated.connect(self.display_backend_warning)
 
-        # Add clickable icon to toggle password visibility to end of box
-        self.showHideAction = QAction(self.tr("Show my passwords"), self)
-        self.showHideAction.setCheckable(True)
-        self.showHideAction.toggled.connect(self.set_visibility)
+        if self.__class__ == AddRepoWindow:
```

Comment:
This is not clean code conforming to the standards of OOP. However the current class hierachy doesn't allow for a different solution. I therefore suggest moving the code for adding the password widgets into a subclass so that `AddRepoWindow` and `ExistingRepoWindow` share a common ancestor with the common code.

Reply:
That would make more sense. Do you want me to push these changes here or open a new PR after this?

Reply:
Regarding the logical error involving `_show_visibility_button`, I think you reviewed an older commit. _toggle_password_visibility it was replaced with `toggle_visibility` public method.

Reply:
> Regarding the logical error involving `_show_visibility_button`, I think you reviewed an older commit. _toggle_password_visibility it was replaced with `toggle_visibility` public method.

No, I only added the comment to the existing conversation but I copied the comment to the new code lines now.